x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation.tsx

```diff
@@ -0,0 +1,175 @@
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <EuiPageBody>
+        <FlashMessages />
+        <EuiPanel>
+          <EuiForm>
+            <form
+              data-test-subj="MetaEngineCreationForm"
+              onSubmit={(e) => {
+                e.preventDefault();
+                submitEngine();
+              }}
+            >
+              <EuiTitle>
+                <EuiText>{META_ENGINE_CREATION_FORM_TITLE}</EuiText>
+              </EuiTitle>
+              <EuiSpacer />
+              <EuiFlexGroup>
+                <EuiFlexItem>
+                  <EuiFormRow
+                    data-test-subj="MetaEngineCreationNameFormRow"
+                    label={META_ENGINE_CREATION_FORM_ENGINE_NAME_LABEL}
+                    helpText={
+                      name.length > 0 && rawName !== name ? (
+                        <>
+                          {SANITIZED_NAME_NOTE} <strong>{name}</strong>
+                        </>
+                      ) : (
+                        ALLOWED_CHARS_NOTE
+                      )
+                    }
+                    fullWidth
+                  >
+                    <EuiFieldText
+                      name="engine-name"
+                      value={rawName}
+                      onChange={(event) => setRawName(event.currentTarget.value)}
+                      autoComplete="off"
+                      fullWidth
+                      data-test-subj="MetaEngineCreationNameInput"
+                      placeholder={META_ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER}
+                      autoFocus
+                    />
+                  </EuiFormRow>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+              <EuiSpacer />
+              <EuiFormRow label={META_ENGINE_CREATION_FORM_ENGINE_SOURCE_ENGINES_LABEL} fullWidth>
+                <EuiComboBox
+                  data-test-subj="MetaEngineCreationSourceEnginesInput"
+                  options={indexedEngineNames.map(engineNameToComboBoxOption)}
+                  selectedOptions={selectedIndexedEngineNames.map(engineNameToComboBoxOption)}
+                  onChange={(options) => {
+                    setSelectedIndexedEngineNames(options.map(comboBoxOptionToEngineName));
+                  }}
+                />
+              </EuiFormRow>
+              <EuiSpacer />
+              {selectedIndexedEngineNames.length > maxEnginesPerMetaEngine && (
+                <EuiCallOut
+                  color="warning"
+                  title={META_ENGINE_CREATION_FORM_MAX_SOURCE_ENGINES_WARNING_TITLE(
+                    maxEnginesPerMetaEngine
+                  )}
+                />
+              )}
+              <EuiSpacer />
+              <EuiButton
+                disabled={
+                  name.length === 0 ||
+                  selectedIndexedEngineNames.length === 0 ||
+                  selectedIndexedEngineNames.length > maxEnginesPerMetaEngine
```

Comment:
Looks like this branch is maybe also uncovered?  Not 100% sure why, I might suggest writing a describe block for checking when the submit button is/isn't disabled

Reply:
Covered!